### PR TITLE
Add org.otfried.Ipe

### PR DESCRIPTION
**Pull request by hmenke**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/otfried/ipe-issues/issues/458
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Ipe is a drawing editor for creating figures in PDF format. It supports making
+small figures for inclusion into LaTeX-documents as well as making multi-page
+PDF presentations.
+
+## Using TeX Live
+
+There is no TeX Live installation bundled with this Flatpak, because it is
+assumed that most users have a Tex Live installation already.  Instead here is
+how to forward an existing TeX Live installation into the sandbox and making Ipe
+aware of where to find it.
```
Comment (hfiguiere) [inline README.md:19]: there is a SDK for that reason. What you are recommending here is to punch a hole in the sandbox.

BTW no user will see this readme.

File: org.otfried.Ipe.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    </p>
+  </description>
+
+  <launchable type="desktop-id">org.otfried.Ipe.desktop</launchable>
+  <categories>
+    <category>Graphics</category>
+    <category>Qt</category>
+  </categories>
+  <content_rating type="oars-1.1"/>
+</component>
```
Comment (hfiguiere) [inline org.otfried.Ipe.metainfo.xml:26]: missing mandatory `screenshots` and `releases`.

File: .github/workflows/update.yml
Comment (hfiguiere) [inline .github/workflows/update.yml:1]: you don't  need this. the `x-checker-data` is handled automatically.

**Comment by hfiguiere** [comment]
bot, build org.otfried.Ipe

File: README.md (hunk 1 above)
Comment (hmenke) [reply README.md:19]: How is this punching a hole in the sandbox? The bind-mounted binaries are going to be run inside the sandbox.

Anyway, I will try to get this working with the TeX Live SDK.

File: org.otfried.Ipe.metainfo.xml (hunk 2 above)
Comment (hmenke) [reply org.otfried.Ipe.metainfo.xml:131]: Two of the historic releases don't have a date in the `news.txt` file and Ipe is not version controlled, so I can't find out when they were released. Also the 6.0 releases have spaces in the version string. Should I just omit anything older than 7.0.0 from the list?

Comment (hfiguiere) [reply org.otfried.Ipe.metainfo.xml:132]: only the current one is required.

File: README.md (hunk 1 above)
Comment (hfiguiere) [reply README.md:19]: any bind mount like that is a hole in the sandbox. 

Also the "requirement" comes against what flatpak provides.

**Comment by hfiguiere** [comment]
bot, build org.otfried.Ipe

File: README.md (hunk 1 above)
Comment (hmenke) [reply README.md]: I respectfully disagree. There is no difference between bind-mounting host binaries into the sandbox and executing them there and bind-mounting an SDK into the sandbox. Nevertheless, I adjusted the README and the SDK solution is now promoted.

**Comment by hfiguiere** [comment]
bot, build bot, build org.otfried.Ipe

**Comment by hfiguiere** [comment]
bot, build org.otfried.Ipe

File: .github/workflows/flatpak.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+name: Flatpak
```
Comment (barthalion) [inline .github/workflows/flatpak.yml:1]: Remember to pause this workflow once it's merged.

File: org.otfried.Ipe.yaml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+runtime-version: 5.15-22.08
+sdk: org.kde.Sdk
+
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --filesystem=xdg-documents:rw
+  - --env=IPELATEXPATH=/app/texlive/bin/x86_64-linux
```
Comment (barthalion) [inline org.otfried.Ipe.yaml:13]: What will happen on aarch64?

Comment (hmenke) [reply org.otfried.Ipe.yaml:13]: Good question. Can I have architecture-dependent `finish-args`?

Comment (hfiguiere) [reply org.otfried.Ipe.yaml:13]: no. 

Either you add both (if it work like PATH), or a wrapper script that set the environment.

**Comment by barthalion** [comment]
bot, build org.otfried.Ipe

**Comment by barthalion** [comment]
/merge @otfried

If this package is considered official, please verify it at https://flathub.org/login when it becomes visible on the website in a few hours.

**Comment by hmenke** [comment]
@barthalion Thank you. Did you invite otfried as maintainer as well?

**Pull request closed without merging**